php/utils/UploadedFileFlowController.php

```diff
@@ -171,7 +171,19 @@ protected function serveCode($site, $fileName, $expires = 0, $restrict_html = fa
     {
         $m = array();
 
-        if (preg_match(";^code/([^/]+)/?(?:/([0-9]+))?(?:(/r/)(.*))?$;", $fileName, $m)) {
+        if (preg_match(
+            "/
+            ^
+            code\/            # Code namespace
+            ([^\/]+)          # Name of file
+            \/?               # Optional further /
+            (?:\/([0-9]+))?   # Number of which code block to get
+            (?:(\/r\/)(.*))?  # /r/ followed by anything - not sure what for
```

Comment:
I believe there's an AJAX callback that uses this method, and this is the domain the request came from. It's not used for this match but is used elsewhere.

Reply:
Makes sense - would be a good idea to document that here once we're sure. (Along with any other weird nuances in any of the other regexes)